src/extension/libjpeg-turbo/src/test/java/org/geoserver/map/turbojpeg/TurboImageWorkerTest.java

```diff
@@ -66,6 +72,44 @@ public void errors() throws IOException {
         }
     }
 
+    @Test
+    public void writeIndexedWithAlpha() throws IOException {
+        if (SKIP_TESTS) {
+            LOGGER.warning(ERROR_LIB_MESSAGE);
+            return;
+        }
+
+        // Create paletted image
+        final byte bb[] = new byte[256];
+        for (int i = 0; i < 200; i++) {
+            bb[i] = (byte) i;
+        }
+        int noDataValue = 200;
+        NoDataContainer noData = new NoDataContainer(noDataValue);
+        final IndexColorModel icm = new IndexColorModel(8, 256, bb, bb, bb);
+        final WritableRaster raster =
+                RasterFactory.createWritableRaster(icm.createCompatibleSampleModel(512, 512), null);
+        for (int i = raster.getMinX(); i < raster.getMinX() + raster.getWidth(); i++) {
+            for (int j = raster.getMinY(); j < raster.getMinY() + raster.getHeight(); j++) {
+                if (i - raster.getMinX() < raster.getWidth() / 2) {
+                    raster.setSample(i, j, 0, (i + j) / 32);
+                } else {
+                    raster.setSample(i, j, 0, 200);
+                }
+            }
+        }
+        // Set no data
+        BufferedImage bi = new BufferedImage(icm, raster, false, null);
+        PlanarImage planarImage = PlanarImage.wrapRenderedImage(bi);
+        planarImage.setProperty(NoDataContainer.GC_NODATA, noData);
```

Comment:
Yep. here it set the nodata property which is taken by the ImageWorker to setup the Alpha:
https://github.com/geotools/geotools/blob/master/modules/library/coverage/src/main/java/org/geotools/image/ImageWorker.java#L1771
So without the fix, it was throwing the IllegalArgumentException